tests/aws/services/events/test_events.py

```diff
@@ -568,6 +568,137 @@ def test_create_connection_validations(self, aws_client):
 
+    @markers.aws.validated
+    def test_put_events_response_entries_order(
+        self, events_put_rule, sqs_create_queue, sqs_get_queue_arn, aws_client, snapshot, clean_up
+    ):
+        """Test that put_events response contains each EventId only once, even with multiple targets."""
+        account_id = aws_client.sts.get_caller_identity()["Account"]
+        region = aws_client.events.meta.region_name
+
+        # Create two SQS queues as targets
+        queue_url_1 = sqs_create_queue()
+        queue_arn_1 = sqs_get_queue_arn(queue_url_1)
+        queue_url_2 = sqs_create_queue()
+        queue_arn_2 = sqs_get_queue_arn(queue_url_2)
+
+        rule_name = f"test-rule-{short_uid()}"
+
+        snapshot.add_transformers_list(
+            [
+                snapshot.transform.key_value("EventId", reference_replacement=False),
+                snapshot.transform.regex(queue_arn_1, "<queue-1-arn>"),
+                snapshot.transform.regex(queue_arn_2, "<queue-2-arn>"),
+                snapshot.transform.regex(rule_name, "<rule-name>"),
+                *snapshot.transform.sqs_api(),
+                *snapshot.transform.sns_api(),
+            ]
+        )
+
+        rule_arn = f"arn:aws:events:{region}:{account_id}:rule/{rule_name}"
+        sqs_policy = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Sid": "EventBridgeToSQS",
+                    "Effect": "Allow",
+                    "Principal": {"Service": "events.amazonaws.com"},
+                    "Action": "sqs:SendMessage",
+                    "Resource": [queue_arn_1, queue_arn_2],
+                    "Condition": {"ArnLike": {"aws:SourceArn": rule_arn}},
+                }
+            ],
+        }
+
+        aws_client.sqs.set_queue_attributes(
+            QueueUrl=queue_url_1, Attributes={"Policy": json.dumps(sqs_policy)}
+        )
+        aws_client.sqs.set_queue_attributes(
+            QueueUrl=queue_url_2, Attributes={"Policy": json.dumps(sqs_policy)}
+        )
+
+        events_put_rule(
+            Name=rule_name,
+            EventPattern=json.dumps(
+                {"source": ["test.source"], "detail-type": ["test.detail.type"]}
+            ),
+        )
+
+        def check_rule_active():
+            rule = aws_client.events.describe_rule(Name=rule_name)
+            assert rule["State"] == "ENABLED"
+
+        retry(check_rule_active, retries=10, sleep=1)
+
+        target_id_1 = f"test-target-1-{short_uid()}"
+        target_id_2 = f"test-target-2-{short_uid()}"
+        target_response = aws_client.events.put_targets(
+            Rule=rule_name,
+            Targets=[
+                {"Id": target_id_1, "Arn": queue_arn_1},
+                {"Id": target_id_2, "Arn": queue_arn_2},
+            ],
+        )
+
+        assert (
+            target_response["FailedEntryCount"] == 0
+        ), f"Failed to add targets: {target_response.get('FailedEntries', [])}"
+
+        test_event = {
```

Comment:
it would make sense to use the constant pattern and detail here:

```
[
            {
                "Source": TEST_EVENT_PATTERN_NO_DETAIL["source"][0],
                "DetailType": TEST_EVENT_PATTERN_NO_DETAIL["detail-type"][0],
                "Detail": json.dumps(EVENT_DETAIL),
            },
        ]
```